machine/machine_core/ssh_connection.py

```diff
@@ -69,6 +70,24 @@ def message(self, *args):
         if self.verbose:
             sys.stderr.write(" ".join(args) + '\n')
 
+    def print_console_log(self):
+        """Prints VM's console to stderr"""
+        if not self.console_file:
+            return
+
+        try:
+            f = open(self.console_file)
+            console_log = f.read().strip()
+        except OSError as ex:
+            sys.stderr.write(f"Failed to open '{self.console_file}': {0}".format(ex.strerror))
```

Comment:
Missing `\n`. Also, please put the exception into the f-string. Combining f-string and .format() is weird.